packages/pros/src/adi/digital.rs

```diff
@@ -0,0 +1,53 @@
+use crate::error::bail_on;
+
+use pros_sys::PROS_ERR;
+
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+pub struct AdiDigitalIn {
+    port: u8,
+}
+
+impl AdiDigitalIn {
+    /// Create an AdiDigitalIn, returning err `AdiError::InvalidPort` if the port is invalid.
+    pub fn new(slot: AdiSlot) -> Result<Self, AdiError> {
+        let port = slot as u8;
+        if port < 1 || port > {pros_sys::NUM_ADI_PORTS as u8} {
+            return Err(AdiError::InvalidPort);
+        }
+        Ok(Self { port })
+    }
+
+    /// Gets the current value of a digital input pin.
+    pub fn new_press(&self) -> Result<bool, AdiError> {
+        Ok(unsafe { bail_on!(PROS_ERR, pros_sys::adi_digital_get_new_press(self.port)) != 0 })
```

Comment:
I think this should take `&mut self` since it's not a pure function in the context of device state. Calling twice doesn't necessarily yield the same result, since PROS internally keeps track of the last time it was called (thus mutating a variable in "`self`").